app/controllers/doorkeeper_applications_controller.rb

```diff
@@ -0,0 +1,84 @@
+class DoorkeeperApplicationsController < ApplicationController
+  before_action :get_application, except: [:index, :new, :create]
+
+  def index
+    @applications = policy_scope(Doorkeeper::Application)
+  end
+
+  def show
+    get_access_token
+  end
+
+  def new
+    authorize Doorkeeper::Application
+    @application = Doorkeeper::Application.new(
+      redirect_uri: "urn:ietf:wg:oauth:2.0:oob",
+      scopes: Doorkeeper.configuration.default_scopes
+    )
+  end
+
+  def edit
+  end
+
+  def create
+    authorize Doorkeeper::Application
+    @application = Doorkeeper::Application.create(application_params.merge(owner: current_user))
+    if @application.valid?
+      redirect_to @application, notice: t(".success")
+    else
+      flash.now[:alert] = t(".failure")
+      render :new
+    end
+  end
+
+  def update
+    generate_token if application_params[:generate_token]
+    @application.update(application_params.except(:generate_token))
```

Comment:
There's no authorization here, so anyone could generate a token for any app?

Reply:
This has actually gone away now, I removed the whole "generate an access token" thing in #3952, better that people just do the OAuth flow properly.